### PR TITLE
json-libgdx - DO NOT MERGE

### PR DESCRIPTION
**Pull request by DaanVanYperen**
DO NOT MERGE - WIP

Json-libgdx passing unit tests, but not integration tests! #319 

Almost there, just need to get some tests fixed.

Some issues:
- [ ] From the users perspective artemis will randomly use either reflection cache when using libgdx. mainly because core io still uses the artemis cache.
- [ ] Several tests failing. (prob. porting issues).
- [ ] Dependency on methods `Reset` and `markSupported`, not available in GDX implementation of `InputStream`. Now we have an overlapping emu declaration of it, prob not desirable.

Edit 1:
- [ ] GWT Build requires 'production' target and loads of memory for some reason, not sure why.
- [ ] Still some errors about Chrome during GWT build, might be because of the IE8 target for unit tests?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -103,6 +102,8 @@
 		<module>artemis-test</module>
 		<module>artemis-cli</module>
 		<module>artemis-gwt</module>
+		<module>artemis-serializer-backends/artemis-json</module>
+		<module>artemis-serializer-backends/artemis-json-libgdx</module>
   </modules>
```
Comment (junkdog) [inline pom.xml]: Should we put a pom in artemis-serializer-backends, let it build its child modules?

Comment (DaanVanYperen) [inline pom.xml]: I mirrored the gdx setup for backends. Couldn't discern a direct need for adding an extra module , if there is one it would make sense!

**Comment by junkdog** [comment]
> Dependency on methods Reset and markSupported, not available in GDX implementation of InputStream.

I was thinking about PR-ing a fix for it today.

File: pom.xml (hunk 1 above)
Comment (junkdog) [inline pom.xml]: Fair enough.

**Comment by DaanVanYperen** [comment]
> I was thinking about PR-ing a fix for it today.

To LibGDX? If you do don't use my InputStream as the PR, I just slapped in some methods without regard of the JDK implementation.

**Comment by DaanVanYperen** [comment]
Edit #1:
- [ ] GWT Build requires 'production' target and loads of memory for some reason, not sure why.
- [ ] Still some errors about Chrome during GWT build, might be because of the IE8 target for unit tests?

**Comment by junkdog** [comment]
> To LibGDX? If you do don't use my InputStream as the PR, I just slapped in some methods without regard of the JDK implementation.

It's core libgdx as well, I think. I haven't confirmed it though, but FIleHandle appears to incorrectly report markSupported even on desktop.

**Comment by DaanVanYperen** [comment]
Btw if you feel like taking a stab at this feel free. ;) branch is up to date.

**Comment by junkdog** [comment]
I could have a look, but i'd most likely be tomorrow or saturday.

OT: serialization on iOS works - I updated the wiki.

**Comment by DaanVanYperen** [comment]
> It's core libgdx as well, I think. I haven't confirmed it though, but FIleHandle appears to incorrectly report markSupported even on desktop.

Ah I see. Well go for it! Feel we're not far from getting all the tests to run on GWT. 

Was all giddy finding io emu in quake 2 for gwt, only to discover libgdx already implements a lot of classes ;)

> I could have a look, but i'd most likely be tomorrow or saturday.

Might put some time in tonight, might take a gaming break ;)

> OT: serialization on iOS works - I updated the wiki.

awesome!

**Comment by junkdog** [comment]
> Might put some time in tonight, might take a gaming break ;)

I've got that Witcher 3 twitching thing in my left eyebrow. It's just a few clicks away from a purchase.

**Comment by DaanVanYperen** [comment]
https://www.reddit.com/r/patientgamers/ I can wait. ;)

**Comment by Namek** [comment]
@junkdog Go for Witcher 3!

**Comment by DaanVanYperen** [comment]
Pfff one wasted night. Looks like dev vs prod mode exposes private/protected methods differently. Can only run tests in prod, but none of my breakpoints work. :p Supposedly should run in superdev, no clue if that is even possible for junit tests.

Btw, looks like I accidentally got one of the reflection caches picked up by the other due to broad netting.

**Comment by junkdog** [comment]
ugh... making progress though. Is dev mode more restrictive than production? Sounds fishy. Could it be a setting somewhere?

**Comment by DaanVanYperen** [comment]
Dev mode runs it in java supposedly. Production runs it in javascript. This has the 'it's something really obvious' smell. Can you give it a try? Maybe you'll notice something I'm overlooking.

**Comment by junkdog** [comment]
Sure (sometime after midnight, going home to sleep after lunch)

**Comment by DaanVanYperen** [comment]
Just ping when you start so we don't overlap effort. ;)

**Comment by junkdog** [comment]
i'll ping you around 1am then

**Comment by DaanVanYperen** [comment]
Revisit after #329

**Pull request closed without merging**